tabix index loading issue

#### Are you using the latest version of [samtools](https://github.com/samtools/samtools/releases/latest) and [HTSlib](https://github.com/samtools/htslib/releases/latest)? If not, please specify.
(run `samtools --version`)
samtools 1.10
Using htslib 1.10.2
Copyright (C) 2019 Genome Research Ltd.

 #### Please describe your environment.
- OS (run `uname -sr` on Linux/Mac OS or ` wmic os get Caption, Version` on Windows)
Darwin 19.3.0
- machine architecture (run `uname -m` on Linux/Mac OS or ` wmic os get OSArchitecture` on Windows)
x86_64
- compiler (run `gcc --version` or `clang --version`)
Apple clang version 11.0.0 (clang-1100.0.33.16)
Target: x86_64-apple-darwin19.3.0
Thread model: posix

#### Please specify the steps taken to generate the issue, the command you are running and the relevant output.
I was trying to retrieve the rs identifier from the genome data on NCBI (ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF) with the following command.

`awk '{gsub("chr","",$1)} {print $1":"$2"-"$3}' coord.txt | parallel --delay 1 tabix ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz {} | awk -v OFS="\t" 'BEGIN {print "chromosome","Coordinate","rsID","Ref","Alt"} {print "chr"$1,$2,$3,$4,$5}'
`

However, the tabix command did not work well and generated the following log issues. 

> [E::hts_idx_load3] Could not load local index file '00-All.vcf.gz.tbi'
> Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
> [E::hts_idx_load3] Could not load local index file '00-All.vcf.gz.tbi'
> Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
> [E::hts_idx_load3] Could not load local index file '00-All.vcf.gz.tbi'
> Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
> [E::hts_idx_load3] Could not load local index file '00-All.vcf.gz.tbi'
> Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
 **[Above Repeated....]**

Here is a small sample data of the "coord.txt" file.

> 1	17375	17375
> 1	17407	17407
> 1	564542	564542
> 1	564621	564621
> 1	564667	564667
> 1	565184	565184
> 1	565292	565292
> 1	565400	565400
> 1	565433	565433
> 1	565697	565697

I want to know how it happens and how to fix it. Thank you very much for your help.

Do I need to clone the entire repository?
When I runs it directly, it outputs the following errors.

fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
make: *** No rule to make target `clean'.  Stop.
make: *** No targets specified and no makefile found.  Stop.
zsh: no such file or directory: ./tabix

--- a/hts.c
+++ b/hts.c
@@ -3441,7 +3441,6 @@ static int idx_test_and_fetch(const char *fn, const char **local_fn, int *local_
 {
     hFILE *remote_hfp;
     FILE *local_fp = NULL;
-    uint8_t *buf = NULL;
     int save_errno;
     htsFormat fmt;
     kstring_t s = KS_INITIALIZE;
@@ -3490,7 +3489,7 @@ static int idx_test_and_fetch(const char *fn, const char **local_fn, int *local_
                 goto fail;
             }
             hts_log_info("Downloading file %s to local directory", fn);
-            buf = (uint8_t*)calloc(buf_size, 1);
+            uint8_t *buf = (uint8_t*)calloc(buf_size, 1);
             if (!buf) {
                 hts_log_error("%s", strerror(errno));
                 goto fail;
@@ -3540,7 +3539,6 @@ static int idx_test_and_fetch(const char *fn, const char **local_fn, int *local_
     save_errno = errno;
     hclose_abruptly(remote_hfp);
     if (local_fp) fclose(local_fp);
-    free(buf);
     free(s.s);
     errno = save_errno;
     return -2;

